providers/implementations/keymgmt/kmgmt_util.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/*
+ * Key Management utility functions to share functionality between the export()
+ * and get_params() methods.
+ * export() uses OSSL_PARAM_BLD, and get_params() used the OSSL_PARAM[] to
+ * fill in parameter data for the same key and data fields.
+ */
+
+#include "internal/kmgmt_util.h"
+#include <openssl/core_names.h>
+
+
+DEFINE_SPECIAL_STACK_OF_CONST(BIGNUM_const, BIGNUM)
+
+int ossl_kmgmt_params_set_int(OSSL_PARAM_BLD *bld, OSSL_PARAM *p,
+                              const char *key, int num)
+{
+    if (bld != NULL)
+        return ossl_param_bld_push_int(bld, key, num);
+    p = OSSL_PARAM_locate(p, key);
+    if (p != NULL)
+        return OSSL_PARAM_set_int(p, num);
+    return 1;
+}
+
+int ossl_kmgmt_params_set_utf8_string(OSSL_PARAM_BLD *bld, OSSL_PARAM *p,
+                                      const char *key, const char *buf)
+{
+    if (bld != NULL)
+        return ossl_param_bld_push_utf8_string(bld, key, buf, 0);
+    p = OSSL_PARAM_locate(p, key);
+    if (p != NULL)
+        return OSSL_PARAM_set_utf8_string(p, buf);
+    return 1;
+}
+
+int ossl_kmgmt_params_set_octet_string(OSSL_PARAM_BLD *bld, OSSL_PARAM *p,
+                                       const char *key,
+                                       const unsigned char *data,
+                                       size_t data_len)
+{
+    if (bld != NULL)
+        return ossl_param_bld_push_octet_string(bld, key, data, data_len);
+
+    p = OSSL_PARAM_locate(p, key);
+    if (p != NULL)
+        return OSSL_PARAM_set_octet_string(p, data, data_len);
+    return 1;
+}
+
+int ossl_kmgmt_params_set_bn_pad(OSSL_PARAM_BLD *bld, OSSL_PARAM *p,
+                                 const char *key, const BIGNUM *bn,  size_t sz)
+{
+    if (bld != NULL)
+        return ossl_param_bld_push_BN_pad(bld, key, bn, sz);
+    p = OSSL_PARAM_locate(p, key);
+    if (p != NULL) {
+        if (sz > p->data_size)
+            return 0;
+        p->data_size = sz;
+        return OSSL_PARAM_set_BN(p, bn);
```

Comment:
Doesn't this skip the padding part of what this function is supposed to do?

Reply:
setting the data_size here achieves this if you look inside OSSL_PARAM_set_BN

Reply:
I thought that the data_size was meant to be _const_ and only the _return_size_ changes.  I think a new function `OSSL_PARAM_set_BN_pad()` is in order.

Reply:
#11390

Reply:
Err what has this PR got to do with this?

Reply:
Put a TODO here for now.